controllers/front/listener.php

```diff
@@ -0,0 +1,69 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Academic Free License 3.0 (AFL-3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/AFL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/AFL-3.0 Academic Free License 3.0 (AFL-3.0)
+ */
+class ps_qualityassurancelistenerModuleFrontController extends ModuleFrontController
+{
+    public function display()
+    {
+        if (!$this->jsEventListenerIsEnabled()) {
+            return;
+        }
+
+        if (Tools::getValue('action') !== 'RecordJSHookExecution') {
+            $this->ajaxDie('Unknown action');
+        }
+
+        $this->processRecordJSHookExecution();
+    }
+
+    public function processRecordJSHookExecution()
+    {
+        $hookName = (string) Tools::getValue('name');
+
+        if (!$hookName) {
+            $this->ajaxRender(
```

Comment:
Create a similar method as it used in the BO and make sure it's a JSON a response.
```
    private function renderJson($data)
    {
        header('Content-Type: application/json');
        $this->ajaxDie(json_encode($data));
    }
```

Reply:
And if it's an error, treat it as error `{"error": "error Message"}`

Reply:
`ajaxDie()` is deprecated while `ajaxRender()` is not. Isn't it better to use `ajaxRender()` then?

Reply:
Yes you can use ajaxRender, but don't forget to die or make sure only the result is returned without nothing else after. And be sure it's a json

Reply:
Fixed in https://github.com/PrestaShop/ps_qualityassurance/pull/12/commits/9e71f01f2a08b7fc527126e0a83d1a7ea458eb0c